src/languages/php.js

```diff
@@ -11,23 +11,101 @@ Category: common
  * @returns {LanguageDetail}
  * */
 export default function(hljs) {
+  const regex = hljs.regex;
   const VARIABLE = {
-    className: 'variable',
+    scope: 'variable',
     begin: '\\$+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*' +
       // negative look-ahead tries to avoid matching patterns that are not
       // Perl at all like $ident$, @ident@, etc.
       `(?![A-Za-z0-9])(?![$])`
   };
+  const CONSTANT = {
+    variants: [
+      {
+        begin: [
+          /const\s+/,
+          /\w+/,
+          /\s*=/,
+        ],
+        beginScope: {
+          1: "keyword",
+          2: "variable",
+        },
+      },
+      {
+        begin: [
+          regex.concat(
+            /::/,
+            regex.lookahead(/(?!class\b)/)
+          ),
+          /(\w+\b(?!\())/
+        ],
+        beginScope: {
+          2: "variable",
+        },
+      },
+      {
+        begin: [
+          regex.concat(
+            /[A-Z]\w*/,
+            /::/,
+            regex.lookahead(/(?!class\b)/)
+          ),
+          /(\w+\b(?!\())/,
+        ],
+        beginScope: {
+          1: "title.class",
+          2: "variable",
+        },
+      },
+      {
+        begin: [
+          /::/,
+          /class/,
+        ],
+        beginScope: {
```

Comment:
```suggestion
        match: [
          /::/,
          /class/,
        ],
        scope: {
```

If there is no `end` please change _all_ of these to match/scope pairs.

Reply:
Fixed everywhere.